dev/archery/archery/docker/core.py

```diff
@@ -378,7 +378,12 @@ def run(self, service_name, command=None, *, env=None, volumes=None,
             else:
                 cmd = service.get('command', '')
                 if cmd:
-                    args.extend(shlex.split(cmd))
+                    # service command might be already defined as a list
+                    # on the docker-compose yaml file.
+                    if isinstance(cmd, list):
+                        cmd = shlex.join(cmd)
+                    else:
```

Comment:
Why the "else"? You must still extend the args after joining the command list...

Reply:
you are correct. I missed to add the command because I couldn't test locally with GPU. I can see the full expanded command now:
```
docker run --rm --gpus all --shm-size 2147483648 -e ARROW_BUILD_STATIC=OFF -e ARROW_CUDA=ON -e ARROW_GANDIVA=OFF -e ARROW_GCS=OFF -e ARROW_ORC=OFF -e ARROW_S3=OFF -e ARROW_SUBSTRAIT=OFF -e ARROW_WITH_OPENTELEMETRY=OFF -e CCACHE_COMPILERCHECK=content -e CCACHE_COMPRESS=1 -e CCACHE_COMPRESSLEVEL=6 -e CCACHE_DIR=/ccache -e CCACHE_MAXSIZE=1G -v /home/raulcd/code/arrow:/arrow -v ubuntu-ccache:/ccache apache/arrow-dev:amd64-ubuntu-20.04-cuda-11.0.3-cpp /bin/bash -c 
  /arrow/ci/scripts/cpp_build.sh /arrow /build &&
  /arrow/ci/scripts/integration_skyhook.sh /build &&
  /arrow/ci/scripts/cpp_test.sh /arrow /build
  ```